### PR TITLE
Fixed 'Extension Manager: display security warning shortcuts correctl…

### PR DESCRIPTION
**Pull request by LarsGit223**
…y #759'.

No more to say. Please check.
Cheers, Lars

### CODE REVIEW

File: helper/repository.php [hunk 1]
```diff
@@ -390,6 +390,10 @@ public function getFilteredPlugins(
             $plugins[$i]['compatible'] = $this->cleanCompat($plugins[$i]['compatible']);
             $plugins[$i]['types']      = $this->listtypes($plugins[$i]['type']);
 
+            if(in_array($plugins[$i]['securitywarning'],$this->securitywarning)){
+                $plugins[$i]['securitywarning'] = $this->getLang('security_'.$plugins[$i]['securitywarning']);
+            }
```
Comment (splitbrain) [inline helper/repository.php]: So when this function always returns the resolved string, I guess resolving at other places like syntax/entry/_showWarnings() is redundant and should be removed?

**Comment by LarsGit223** [comment]
NO! Please do not remove it. It is not redundant.

This is the "call scenario" for the _showWarnings function:
:arrow_forward: User views plugin page XYZ on dokuwiki.org or any other DokuWiki server/installation
:arrow_down: Tag "---- plugin ----" is recognized
:arrow_down: function render is called
:arrow_down: calls _showData
:arrow_right: calls _showWarnings

This is the "call scenario" for the getFilteredPlugins function:
:arrow_forward: User views extension manager on dokuwiki.org or any other DokuWiki server/installation
:arrow_down: Extension manager sends http-request to dokuwiki.org to fetch the data
:arrow_down: fetch is served by api.php
:arrow_right: calls getFilteredPlugins

:exclamation: If you compare the output created for the security warning you might also notice that in function getFilteredPlugins the security link is not added. The function has got no access to a renderer instance so I could not create the security link.

**Comment by splitbrain** [comment]
Ah I see. I wonder if it would be sensible to move the security resolving into it's own helper function so we avoid the duplicate code.

**Comment by LarsGit223** [comment]
Done.

**Pull request merged**